### DISCUSSION TITLE
Tracking Realtime Message Count

### DISCUSSION THREAD

**Discussion by joshuadunning** [supabase/supabase#21120] [Questions]
Hello, we have an app that had a giant spike in users, but also had a giant spike in messages to ~20m messages / day. There is obviously a memory leak of some sort causing the spike from 500k messages to 20m, but there is no way to get insight into the messages being sent. 

It seems that I'm being billed for a metric that I am not able to track and debug. To properly debug, I need to see things like how many messages are being generated in a channel in a time frame. 

I have multiple support tickets out with lackluster response times to an issue burning money in our pockets right now. Anyone have any insight on how to track if our changes have any affect besides waiting 24 hours for the usage dashboard to update?

**Comment by akabutnicer** [comment]
Every time a message is sent, make a counter go up by one. I'd say limit it to 15k at the max, then once the limit is reached block any realtime broadcasting for that user. Or the server in general.

**Reply by akabutnicer** [reply]
The example of this can be:
```javascript
var client = createClient(URL, apikey);
var channel = client.channel('main', {
     config: {
          broadcast: { ack: true }
     }
});
var serverMessages = 0;

function sendMessage(event, data) {
   return new Promise((resolve) => channel.send({ type: 'broadcast', event, payload: data }).then(() => resolve(++serverMessages)));
}

if (serverMessages < 15000) {
      sendMessage('message', { text: 'hello' }).then(result => console.log(result));
}
else {
     supabase.auth.updateUser({
             user_metadata: {
                     rate_limited: true,
              }
    })
}
